proposals/001-controlled-rollout.md

```diff
@@ -0,0 +1,115 @@
+# Controlled Rollout proposal
+
+## Why
+
+New features are great, everybody loves them. However, new features come by the hand of new code. New code can have bugs 
+or it can have performance regressions.
+
+Features aren't perfect for all users. Sometimes they are perfect for some users and not others. Sometimes they are 
+perfect for some hosts and not others.
+
+Rolling out a feature shouldn't always be a binary choice: enabled/disabled. In an ideal world, all features would be 
+enabled by default, everybody would love them all, and they would work flawlessly for all possible use cases.
+
+We are in the real world, though, which is not ideal. So we should give people running Fleet tools to rollout features 
+slowly, so that they can update infrastructure if needed, or only use a feature within the scope that is useful for 
```

Comment:
I think this is a great idea, should we put guidelines or a DRY individual to make the final decision regarding which features can be configured?

I can imagine this feature left unused because of the extra work, or being overused and suddenly everything is behind a config.

Reply:
It'll be work to strike a balance. The default is probably NOT to use this unless there's a need.